apps/shell/tash_command.c

```diff
@@ -433,46 +433,67 @@ int check_exclam_cmd(char *buff)
 
 	/* Find the '!' in the input character */
 
-	while ((exclam_ptr = strchr(buff, ASCII_EXCLAM)) != NULL) {
+	int qc = 0;		/* Flag to check whether ! is inside quotation mark */
```

Comment:
Instead of using a integer flag, use boolean flag, since it's it's either true or false. Improves code readability.